Source/Core/Core/PowerPC/JitArm64/JitAsm.cpp

```diff
@@ -654,5 +654,29 @@ void JitArm64::GenerateCommonAsm()
   paired_store_quantized[30] = storeSingleS8Slow;
   paired_store_quantized[31] = storeSingleS16Slow;
 
-  GetAsmRoutines()->mfcr = nullptr;
+  // Input in X2, output in W3, clobbers X2-X5 and flags. X0 and X1 are not used, because this is
+  // called from code in JitArm64_Backpatch.cpp which uses those two registers for other purposes.
+  cdts = GetCodePtr();
+
+  UBFX(X4, X2, 52, 11);
+  SUB(W5, W4, 874);
+  CMP(W5, 896 - 874);
+  LSR(X3, X2, 32);
+  FixupBranch denormal = B(CCFlags::CC_LS);
+
+  AND(X3, X3, 34, 1, true);  // 0xc0000000
```

Comment:
Just use the ANDI2R. Indeed a pointless function call, but it makes sure that the comment behind it is actually correct.

Reply:
Actually, ANDI2R failed to construct this without a scratch register. I should probably investigate why at some point. (The logic in the function is a little complicated...)

Reply:
Nevermind, I tried it again and it works fine.